src/cpu.rs

```diff
@@ -527,6 +549,181 @@ impl Cpu {
         self.condition_codes.set_zero(self.registers.a);
         self.condition_codes.set_parity(self.registers.a);
     }
+
+    // arithmetic group
+    fn add(&mut self, val: u8) {
+        // val is value containted in register
+        let reg_a = self.registers.a;
+        let res: u16 = (reg_a as u16).wrapping_add(val as u16);
+        // put result in accumulator
+        self.registers.a = res as u8;
+        // update flags
+        self.condition_codes.set_zero(res as u8);
+        self.condition_codes.set_sign(res as u8);
+        self.condition_codes.set_parity(res as u8);
+        self.condition_codes.set_carry(res > 0xFF);
+        self.condition_codes
+            .set_aux_carry((reg_a & 0x0F) + (val & 0x0F) > 0x0F);
```

Comment:
In my naming of hex values, I didn't prepend the values with a 0 for one byte values. I don't have a preference here, so I can change mine to include the 0 or we can remove them from yours for consistency. Either way, just let me know.

Reply:
Ah yes I noticed that, I was just going by my default. I can changes these since I'm already going to be editing a few things.